src/utils/logger.js

```diff
@@ -1,17 +1,52 @@
-import bunyan from 'bunyan'
+import pino from 'pino'
 import { getLogLevel, getPackageJson } from '../config.js'
 
 let logger
 const packageJson = await getPackageJson()
 
 export default function getLogger() {
-  if (logger === undefined)
-    logger = bunyan.createLogger({
-      name: `${packageJson.name}-${packageJson.version}`,
-      stream: process.stdout,
-      level: getLogLevel(),
-      serializers: bunyan.stdSerializers,
-    })
-
+  if (logger) return logger
+  logger = pino({
+    name: `${packageJson.name}-${packageJson.version}`,
+    level: getLogLevel(),
+    depthLimit: 10, // default: 5
+    redact: redactSensitiveInformationFromLogs(),
+  })
   return logger
 }
+
+function redactSensitiveInformationFromLogs() {
+  const fieldsToRedact = [
+    // payment specific fields
+    'customer',
+    'transactions',
+    'interfaceId',
+    'interfaceInteractions',
+    'paymentStatus',
+    // order specific fields
+    'customerId',
+    'customerEmail',
+    'customerGroup',
+    'anonymousId',
+    'shippingAddress',
+    'billingAddress',
+    'country',
+    'paymentInfo',
+    'shippingInfo',
+    'itemShippingAddresses',
+    'orderState',
+    'shipmentState',
+    'paymentState',
+    'lineItem[*].shippingDetails',
+    'customLineItems[*].shippingDetails',
```

Comment:
`interfaceId` - you seem right, but I thought this might be some info that could indirectly represent a payment.
`anonymousId` - I thought that this could be anything or external token kind of session information.
`customerId` - it's unique to a customer I guess with the log you could access the customer details, I thought this is indirect and could `relate to` a specific customer.
`customerGroup`- similar to customerId.
`states` - I thought those are specific info that could give indirect details of an order.

But in general maybe we are redacting to much info here, so I will remove those for now.

Reply:
https://github.com/commercetools/commercetools-subscriptions/pull/25/commits/3fed8e4a3bf40567aec73055c20f6194929ce50e